### ISSUE TITLE
Received routes have unusual next hop

### ISSUE THREAD

**Issue by pshemk** [adrienverge/openfortivpn#616]
When I allow the routes to be installed with `set-routes = 0` the next hop for those routes ends up being the IP address of the `ppp0` itself, for example:
```
10.0.0.0/8 via 10.200.128.20 dev ppp0 scope link
```
Is there a way to make it look like this:
```
10.0.0.0/8 dev ppp0 scope link
```

The reason I'm trying to change them is so I can redistribute them using `bird` into BGP, but having the next hop as interface IP breaks it.

**Comment by mrbaseman** [comment]
@pshemk you are working on Linux, right?
We had to implement this behavior a long time ago for the port to Mac OSX and FreeBSD. 
Somehow I was assuming that we need this hop on Linux, but it seems not to be the case.
I have created [this branch](https://github.com/mrbaseman/openfortivpn/tree/routing) and #617 
Could you test this in your environment, please?

**Comment by pshemk** [comment]
Hmm, after compiling that version and testing it I ended up with two sets of routes when the tunnel is up:
```
10.0.0.0/8 via 10.200.128.3 dev ppp0 scope link                                                                                                                                                
10.0.0.0/8 dev ppp0 scope link                                                                                                                                                                 
```
is that to be expected? The setup actually works for me, but might be a bit confusing for some.

**Comment by mrbaseman** [comment]
No, that's not expected. The previous connection was closed properly, so that the routes weren't still there, maybe cached somehow and magically reappearing as soon as the interface is up again?
Do you run openfortivpn "by hand" from the command line? Just to rule out issues caused by Network Manager or the nm-plugin...

**Comment by pshemk** [comment]
I run this instance inside a docker container. After a fresh creation of the container and simply running the openfortivpn:
```
$ sudo docker exec -it fortivpn /usr/bin/openfortivpn
INFO:   Connected to gateway.
INFO:   Authenticated.
INFO:   Remote gateway has allocated a VPN.
Using interface ppp0
Connect: ppp0 <--> /dev/pts/1
INFO:   Got addresses: [10.200.128.3], ns [10.200.2.15, 10.200.2.16]
INFO:   negotiation complete
INFO:   negotiation complete
local  IP address 10.200.128.3
remote IP address 192.0.2.1
primary   DNS address 10.200.2.15
secondary DNS address 10.200.2.16
INFO:   Interface ppp0 is UP.
INFO:   Setting new routes...
INFO:   Adding VPN nameservers...
INFO:   Tunnel is up and running.
```

A peek at the routing table:
```
$ docker exec -it fortivpn ip r sh | grep 10.0
10.0.0.0/8 via 10.200.128.3 dev ppp0 scope link
10.0.0.0/8 dev ppp0 scope link
```

If I kill the process all routes disappear as expected and point to the default gw:
```
$ docker exec -it fortivpn ip r sh 10.0.0.0/8
[empty answer]
$ docker exec -it fortivpn ip r g 10.0.0.0/8
10.0.0.0 via 192.168.101.1 dev eth0 src 192.168.101.2 uid 0
    cache
```

When running in verbose mode:
```
DEBUG:  Route not found.
DEBUG:  ip route show to a.b.44.6/255.255.255.255 dev !ppp0
DEBUG:  Setting route to vpn server...
DEBUG:  ip route show to a.b.44.6/255.255.255.255 via 192.168.101.1 dev eth0
DEBUG:  ip route add to a.b.44.6/255.255.255.255 via 192.168.101.1 dev eth0
DEBUG:  ip route add to 10.0.0.0/255.0.0.0 via 10.200.128.3 dev ppp0
```

What should be the reported version, just in case I got the wrong repo?
```
$ sudo docker exec -it fortivpn /usr/bin/openfortivpn --version
1.13.3
```

**Comment by pshemk** [comment]
Just to clarify - there's only a handful of processes running, so I don't think anything in the background could store those routes and inject them back:
```
$ docker exec -it fortivpn ps axfww
  PID TTY      STAT   TIME COMMAND
  231 pts/2    Rs+    0:00 ps axfww
  189 pts/0    Ss+    0:00 /bin/bash
  200 pts/0    Sl     0:00  \_ /usr/bin/openfortivpn
  201 pts/1    Ss+    0:00      \_ /usr/sbin/pppd 115200 :192.0.2.1 noipdefault noaccomp noauth default-asyncmap nopcomp receive-all nodefaultroute nodetach lcp-max-configure 40 mru 1354 usepeerdns logfd 2
    1 ?        Ss     0:07 /usr/bin/python /usr/bin/supervisord -c /etc/supervisor/supervisord.conf
    8 ?        S      0:00 python /kill_supervisor.py
   10 ?        S      0:00 /usr/sbin/bird6 -f -d -s /tmp/bird6.ctl -P /tmp/bird6.pid
  114 ?        S      0:01 /usr/sbin/bird -f -d -s /tmp/bird.ctl -P /tmp/bird.pid
```

**Comment by mrbaseman** [comment]
So far we have incremented the version when we have finalized the releases. I have added a revision string to my branch. It's the Id of the last commit in git. That's currently `808848d` in [this](https://github.com/mrbaseman/openfortivpn/commits/routing) branch and it is shown when you run `openfortivpn -v`

**Comment by pshemk** [comment]
I've updated to that version and the revision is showing fine. That version has also resolved my problem with double routes as well. 

```
$ docker exec -it fortivpn openfortivpn -v
DEBUG:  openfortivpn 1.13.3
DEBUG:      revision 808848d
```

```
$ docker exec -it fortivpn ip r sh | grep 10.0
10.0.0.0/8 dev ppp0 scope link
```

**Comment by mrbaseman** [comment]
We have changed the format of the revision output (see #627). 
#617 is not merged yet, but I have rebased the commits on the current master.

**Comment by mrbaseman** [comment]
I have merged #617 in the master branch now, so it will be in the next release
